cli/create.go

```diff
@@ -189,6 +191,11 @@ func (r *RootCmd) create() *clibase.Cmd {
 			Description: "Specify a duration after which the workspace should shut down (e.g. 8h).",
 			Value:       clibase.DurationOf(&stopAfter),
 		},
+		clibase.Option{
+			Flag:        "build-options",
+			Description: "Prompt for one-time build options defined with ephemeral parameters.",
+			Value:       clibase.BoolOf(&buildOptions),
+		},
```

Comment:
What's the rationale behind this? I thought we would always prompt for ephemeral parameters? In this case, if we do not speciify `--build-options` and the template defines ephemeral parameters, would the build not just fail?

Reply:
From a user perspective, they are [optional](https://github.com/coder/coder/issues/6828#issuecomment-1623843335). Workspace owner can either start the workspace or start with options (= ephemeral parameters). It is expected not to ask for ephemeral values every time.

> What's the rationale behind this? I thought we would always prompt for ephemeral parameters? In this case, if we do not speciify --build-options and the template defines ephemeral parameters, would the build not just fail?

They are still parameters, so Coder will switch to default values for them. If the user defines `default: ""`, then Coder will assume an empty string.

Ephemerality just prevents Coder from passing values between workspace builds. Everything else works for parameters the same way.

Reply:
Ah, I misunderstood. 👍